crewai_tools/tools/contextual_tool/contextual_tool.py

```diff
@@ -0,0 +1,486 @@
+from typing import Any, Optional, Type, List, Union
+from crewai.tools import BaseTool
+from pydantic import BaseModel, Field
+import time
+import requests
+import warnings
+
+
+class ContextualQuerySchema(BaseModel):
+    """Schema for contextual query tool."""
+    query: str = Field(..., description="Query to send to the Contextual AI agent.")
+
+
+class ContextualTool(BaseTool):
+    """Tool to interact with Contextual AI RAG agents."""
+    
+    name: str = "Contextual AI Query Tool"
+    description: str = "Use this tool to query a Contextual AI RAG agent with access to your documents"
+    args_schema: Type[BaseModel] = ContextualQuerySchema
+    
+    api_key: str
+    agent_id: Optional[str] = None
+    datastore_id: Optional[str] = None
+    contextual_client: Any = None
```

Comment:
I see your tool requires extra dependencies. I'd recommend adding attributes to map the environment variables and package dependencies.

[Here's a reference](https://github.com/crewAIInc/crewAI-tools/blob/main/crewai_tools/tools/ai_mind_tool/ai_mind_tool.py#L35)